packages/@uppy/core/src/Uppy.ts

```diff
@@ -1653,7 +1655,7 @@ export class Uppy<M extends Meta, B extends Body> {
    */
   use<T extends typeof BasePlugin<any, M, B>>(
     Plugin: T,
-    opts?: ConstructorParameters<T>[1],
+    ...args: OmitFirstArg<ConstructorParameters<T>>
```

Comment:
why are we spreading all other arguments? there will only ever be two arguments of which the second `opts`?

Reply:
1. We don't know if there ever going to be two arguments, we could imagine a user plugin having more than two arguments
2. I think it's the only way to make TS work?

Reply:
OK but how?

Reply:
There is no PR description and I still don't understand why we need this if we don't want to support a third argument. Can you help us understand?

Reply:
This PR is not trying to support third arguments, that's just a happy accident. Does the modification in the tests make it clearer? I want TS to report an error when `opts` is defined as mandatory in the plugin constructor and missing from the `.use` call.

Reply:
I see now 👍 

Can we add some docs to the `OmitFirstArg` type so people know what it does and how it can be used and a comment on line 1658 to explain _why_ we need it? I'd like to make it a habit to explain hard to understand types in public APIs for maintenance.